packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -775,6 +775,35 @@ export class QueryModel {
         };
     }
 
+    get showImportDataButton(): boolean {
+        return this.queryInfo?.showInsertNewButton && this.queryInfo.importUrl && !this.queryInfo.importUrlDisabled;
+    }
+
+    get showInsertNewButton(): boolean {
+        return this.queryInfo?.showInsertNewButton && this.queryInfo.insertUrl && !this.queryInfo.insertUrlDisabled;
+    }
```

Comment:
I think we should move the implementation of these methods to `QueryInfo`. These methods seem specific to `QueryInfo`, they are computed only using information from the `QueryInfo`. We can't make it a `get` method on `QueryInfo` because it clashes with the property name, but we could create a `getShowImportDataButton` method that takes into account `importUrl` and `importUrlDisabled`. If you still want a pass-through for convenience that is fine (for example `get schemaName` is just a pass-through to the `schemaQuery.schemaName` attribute). If you keep the pass-through methods on `QueryModel` definitely keep them `get` methods though so they can be accessed as properties.

Reply:
Do we have criteria for what should be in QM and what in QueryInfo?  Looks like there are several getters in QM just passing through data from QueryInfo with minimal manipulation.

Reply:
Yes, if you look at the other methods that are effectively pass-throughs, they are taking information that is _not_ stored on the `QueryInfo`, but exists only on the `QueryModel` and passing it to the method on `QueryInfo` in order to get a desired behavior. In other cases we are transforming the response from the `QueryInfo` in some way, mostly to not return ImmutableJS objects. In this case, the behavior you want can be entirely determined by state on the `QueryInfo` and requires no state on the `QueryModel`, it can be entirely isolated, and should be. I do think there would still be utility in having a pass-through method on the `QueryModel`, much like we do with `get keyColumns`, `get schemaName`, etc.